src/tfr_control/src/controller_launcher.cpp

```diff
@@ -0,0 +1,49 @@
+/**
+ * controller_launcher.cpp
+ * 
+ * This is the primary node for the control package, it registers the controller
+ * instance with the controller manager and handles calling updates on the
+ * hardware.
+ */
+#include <ros/ros.h>
+#include <sstream>
+#include <controller_manager/controller_manager.h>
+#include "controller.h"
+
+int main(int argc, char **argv)
+{
+    // "controller_launcher is the name of this node at run-time"
+    ros::init(argc, argv, "controller_launcher");
+    ros::NodeHandle n;
+
+    // The controller that we'll be registering with the controller_manager
+    tfr_control::Controller controller;
+
+    // Register our controller with the controller_manager
+    controller_manager::ControllerManager cm(&controller);
+
+    // Start a spinner with one thread, we don't need any more than that
+    // This is actually required (vs calling ros::spinOnce() in the loop)
+    // by ControllerManager for some reason, as I found in testing.
```

Comment:
probably because controller manager is threaded and multiplexes between multiple controllers.

Reply:
That and I think cm.update actually hangs on waiting for a topic update or something, so it needs a spinner on another thread to do it?